exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -243,17 +253,23 @@ func (prwe *PrwExporter) export(ctx context.Context, tsMap map[string]*prompb.Ti
 
 	httpResp, err := prwe.client.Do(httpReq)
 	if err != nil {
-		return err
+		return consumererror.Permanent(err)
 	}
 
+	// Reference for different behavior according to status code:
+	// https://github.com/prometheus/prometheus/pull/2552/files#diff-ae8db9d16d8057358e49d694522e7186
 	if httpResp.StatusCode/100 != 2 {
 		scanner := bufio.NewScanner(io.LimitReader(httpResp.Body, 256))
 		line := ""
 		if scanner.Scan() {
 			line = scanner.Text()
 		}
 		errMsg := "server returned HTTP status " + httpResp.Status + ": " + line
-		return errors.New(errMsg)
+		if httpResp.StatusCode >= 500 && httpResp.StatusCode < 600 {
+			return errors.New(errMsg)
```

Comment:
I think you want permanent in this case not the other, can you add test?

Reply:
I thought we should retry if its a 503 or other 500 errors.  [This](https://github.com/prometheus/prometheus/pull/2552/files#diff-ae8db9d16d8057358e49d694522e7186) is the Prometheus implementation I referenced. 

I don't know how I can make the proto buf marshal fail or the construction of a http request return error.

Reply:
Ok, if you want to retry on 5xx add a comment to the code that this is expected based on link to prom implementation.

Reply:
Cool. I have added comments from 259 - 262